mdata/chunk/archive/archive.go

```diff
@@ -0,0 +1,69 @@
+package archive
+
+import (
+	"bytes"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"github.com/raintank/metrictank/mdata/chunk"
+	"gopkg.in/raintank/schema.v1"
+)
+
+//go:generate msgp
+type Archive struct {
+	RowKey          string
+	SecondsPerPoint uint32
+	Points          uint32
+	Chunks          []chunk.IterGen
+}
+
+//go:generate msgp
+type Metric struct {
+	MetricData        schema.MetricData
```

Comment:
Right. I guess one could argue it's better to use `MetricDefinition` in this struct because it gets sent over the network and that datatype would be slightly smaller, on the other hand I assume that most of the additional `0`s in `MetricData` would get eaten by the compression anyway.